front/etf-recommendation/app/api/history/[symbol]/route.ts

```diff
@@ -0,0 +1,44 @@
+// app/api/history/[symbol]/route.ts
+import { NextResponse } from 'next/server';
+import yahooFinance from 'yahoo-finance2';
+import { subDays, subMonths, format } from 'date-fns';
+
+type RangeKey = '7d' | '30d' | '3mo' | '6mo' | '1y';
+
+function calcOpts(rng: RangeKey) {
+    const end = new Date();
+    let start: Date;
+    switch (rng) {
+        case '7d':  start = subDays(end, 7);   break;
+        case '30d': start = subDays(end, 30);  break;
+        case '3mo': start = subMonths(end, 3); break;
+        case '6mo': start = subMonths(end, 6); break;
+        case '1y':  start = subMonths(end, 12);break;
+        default:    start = subDays(end, 30);
+    }
+    return {
+        period1: format(start, 'yyyy-MM-dd'),
+        period2: format(end,   'yyyy-MM-dd'),
+        interval: '1d' as const,
+    };
+}
+
+export async function GET(
+    request: Request,
+    { params }: { params: Promise<{ symbol: string }> }  // ★ Promise로 선언
+) {
+    const { symbol } = await params;  // ★ await 처리
+    const rng = (new URL(request.url)).searchParams.get('range') as RangeKey || '30d';
```

Comment:
[공식문서](https://nextjs.org/docs/app/building-your-application/routing/route-handlers#url-query-parameters)에 따르면 다음과 같은 방식이어야 할 것 같네요.
```suggestion
export function GET(request: NextRequest) {
  const searchParams = request.nextUrl.searchParams
  const rng = searchParams.get('range') as RangeKey || '30d';
}
```